buildSrc/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-instrumentation.gradle.kts

```diff
@@ -1,12 +1,15 @@
 plugins {
   id("io.opentelemetry.instrumentation.javaagent-testing")
   id("io.opentelemetry.instrumentation.muzzle-check")
-  id("io.opentelemetry.instrumentation.javaagent-codegen")
+  id("io.opentelemetry.instrumentation.muzzle-generation")
 }
 
 dependencies {
   add("muzzleBootstrap", "io.opentelemetry.instrumentation:opentelemetry-instrumentation-api-annotation-support")
   add("muzzleBootstrap", "io.opentelemetry.javaagent:opentelemetry-javaagent-instrumentation-api")
   add("muzzleTooling", "io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api")
   add("muzzleTooling", "io.opentelemetry.javaagent:opentelemetry-javaagent-tooling")
+  add("muzzleTooling", "org.slf4j:slf4j-simple")
+
+  add("codegen", "io.opentelemetry.javaagent:opentelemetry-javaagent-tooling")
```

Comment:
Can you add some comments to describe the dependencies and where they are used? In particular helpful to describe the difference between this dependency and the one in build.gradle, which I guess are related somehow

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/gradle-plugins/build.gradle.kts#L24

Reply:
PTAL